### PR TITLE
Add rundeck cli and project management

### LINKED ISSUE #67: Add support for adding readme.md and motd.md for projects.
New in version 2.5

### LINKED ISSUE #181: Change project define to be backend-independent
Now that #167 is merged, the project define should be changed to use rd-project to create its projects.

https://rundeck.org/2.6.2/man1/rd-project.html

### LINKED ISSUE #449: rundeck project type/provider
We should be able to create a project using db storage type instead of file system. (https://docs.rundeck.com/docs/administration/projects/configuration.html)

`rundeck::config::project` permit only filesystem based projects definition

### PR DESCRIPTION
**Pull request by Joris29**
#### Pull Request (PR) description
This introduces the rundeck cli as a seperate class and als add a define to create and manage projects

Fixes #449
Fixes #181
Fixes #67

### CODE REVIEW

**Comment by Joris29** [comment]
@kenyon Hello it's been a while but i wanted to propose this first draft of the changes i added.
If you have questions or proposals feel free to add them.

File: data/Debian.yaml [hunk 1]
```diff
@@ -9,3 +9,5 @@ rundeck::repo_config:
     key:
       name: rundeck
```
Comment (kenyon) [inline data/Debian.yaml:10]: This should fix the repo signature validation problem (should be a separate PR, but I noticed it here):

```suggestion
      name: rundeck.asc
```

**Review by kenyon — COMMENTED** [review]
The new code looks fine, but looks like we need to get the tests passing.

File: data/Debian.yaml (hunk 1 above)
Comment (Joris29) [reply data/Debian.yaml:10]: Alright i will add some tests for the new code and create a separate PR for repo thing

**Comment by Joris29** [comment]
@kenyon Added unit tests and improved some old test also changed the cli class itself a bit but still having issues with ubuntu repo gpg check

**Comment by Joris29** [comment]
@kenyon I Updated some things and had to remove CentOS 7 from unit tests as jq is not available by default.
Updated refs and readme i also created clean separate commits so those can added as is i think.
This only supports project config itself not project resources (yet). I will this i a follow up PR but i don't want to make it to big.

File: manifests/init.pp [hunk 2]
```diff
@@ -106,6 +111,18 @@
 #   Allows you to use your own override template instead to config rundeckd init script.
 # @param service_script
 #   Allows you to use your own override template instead of the default from the package maintainer for rundeckd init script.
+# @param manage_cli
+#   Wheter to manage rundeck cli config and resource with the rundeck class or not.
```
Comment (kenyon) [inline manifests/init.pp:115]: typo fix

```suggestion
#   Whether to manage rundeck cli config and resource with the rundeck class or not.
```

**Review by kenyon — COMMENTED** [review]
found one typo, otherwise looks good to me

**Comment by Joris29** [comment]
Fixed the typo
